kytos/core/helpers.py

```diff
@@ -405,8 +389,10 @@ def validate_openapi_request(
     if body:
         content_type_json_or_415(request)
     openapi_request = StarletteOpenAPIRequest(request, body)
-    result = openapi_request_validator.validate(spec, openapi_request)
-    _request_validation_result_or_400(result)
+    try:
+        unmarshal_request(openapi_request, spec)
```

Comment:
Notice that with `unmarshal_request` it has become harder here, which also required you to access upstream private attributes with double underscore (dunder) which isn't a good sign, we do have the version pinned so it won't back fire immediately, but relying on internal single underscore or dunder attrs will make this harder to maintain. 

On `openapi-core` 0.18.2, looks like they are still [maintaining a similar validation functionality](https://github.com/python-openapi/openapi-core/blob/0.18.2/openapi_core/shortcuts.py#L277), as we used to use that returns a [result](https://github.com/python-openapi/openapi-core/blob/0.18.2/openapi_core/unmarshalling/datatypes.py#L9) instead of an exception.  

https://openapi-core.readthedocs.io/en/latest/validation.html#request-validation

Check out if you can refactor this and use a `validate_request(request, spec)` method. Let me know if you've also tried and encountered any other blocker